types/ffmpeg.js/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// Type definitions for package ffmpeg.js 3.1
+// Project: https://github.com/Kagami/ffmpeg.js
+// Definitions by: Vladimir Grenaderov <https://github.com/VladimirGrenaderov>,
+//                 Max Boguslavskiy <https://github.com/maxbogus>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+export interface Options {
+    arguments: string[];
+    MEMFS?: Video[];
+    print?(data: any): void;
+    printErr?(data: any): void;
+    onExit?(code: unknown): void;
+    stdin?(data: any): void;
+    mounts?: Mount[];
+    TOTAL_MEMORY?: number;
+}
+
+export interface Opts {
+    root: string;
+}
+
+export interface Mount {
+    type: string;
+    opts: Opts;
+    mountpoint: string;
+}
+
+export interface Result {
+    MEMFS: Video[];
+}
+
+export interface Video {
+    data: Uint8Array;
+    name: string;
+}
+
+export namespace Worker {
+    interface Data {
+        type: string;
+        data: string;
+    }
+
+    interface PostMessageOptions {
+        type: string;
+        arguments: string[];
+    }
+
+    interface OnMessageOptions {
+        data: Data;
+    }
+}
+
+export function ffmpeg(opts: Options): Result;
```

Comment:
It seems like the [underlying library](https://www.npmjs.com/package/ffmpeg.js) is not using `export default`, so this .d.ts file should not either. Instead, you should follow the appropriate template from [the TypeScript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/templates.html). Note that if you want to use `import name from 'package';` in your consuming code, you can use the `esModuleInterop` compiler setting.

Reply:
Fixed

Reply:
This should also have `export =`.

Regarding `Worker`, it looks like you should already get type info for this from `lib.dom.d.ts`, so you don't need to export it (same goes for the other files).

Reply:
I rewrote export so it match parent library. Thanks for pointing on that problem.